sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -161,18 +161,52 @@ public HttpTransport(
    */
   @Override
   public boolean isRetryAfter(final @NotNull String type) {
-    if (sentryRetryAfterLimit.containsKey(type)) {
-      final Date date = sentryRetryAfterLimit.get(type);
+    final String category = getCategoryFromType(type);
+    final Date currentDate = new Date();
 
-      return !new Date().after(date);
-    } else if (sentryRetryAfterLimit.containsKey(HTTP_RETRY_DEFAULT_CATEGORY)) {
-      final Date date = sentryRetryAfterLimit.get(HTTP_RETRY_DEFAULT_CATEGORY);
+    // check all categories
 
-      return !new Date().after(date);
+    final Date dateAllCategories = sentryRetryAfterLimit.get(HTTP_RETRY_ALL_CATEGORIES);
+    if (dateAllCategories != null) {
+      if (!currentDate.after(dateAllCategories)) {
+        return true;
+      }
+    }
+    // check for specific category
+    final Date dateCategory = sentryRetryAfterLimit.get(category);
+    if (dateCategory != null) {
+      return !currentDate.after(dateCategory);
     }
+
     return false;
   }
 
+  /**
+   * Returns a rate limiting category from type
+   *
+   * @param type the type (eg event, session, attachment, ...)
+   * @return the category eg (error, session, attachment)
+   */
+  private @NotNull String getCategoryFromType(final @NotNull String type) {
+    // fallback everything else to 'default' category
```

Comment:
`h`: Hm the implementation on Cocoa is now different: https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentryRateLimitCategoryMapper.m
Is this now the proper way now @jan-auer ?

Reply:
Okay I see the internal docs state :`default`: Events with an event_type not listed explicitly below.

Reply:
I've added all the known types and used by either this or SDKs that bundle sentry-android, which can be sentry-react-native or sentry-native, so I believe this should be the correct implementation